### PR TITLE
Update for new Visibilities object and some tidy

### PR DESCRIPTION
**Pull request by samaloney**
* Upate internal code for new vis object
* Update imaging demo to use new vis object feature (slice, meta)
* Small tidy ups

### CODE REVIEW

File: examples/imaging_demo.py [hunk 1]
```diff
@@ -213,52 +197,68 @@
...
+header = make_fitswcs_header(
+    clean_map.data,
+    max_stix.transform_to(Helioprojective(obstime=vis_tr.center, observer=solo)),
+    telescope="STIX",
+    observatory="Solar Orbiter",
+    scale=pixel,
+    rotation_angle=90 * u.deg + roll,
+)
+
+clean_map = Map((clean_map.data, header))
```
Comment (paolomassa) [inline examples/imaging_demo.py:234]: Why is the map created outside of clean? Is it possible to make it inside as for the other methods?

File: examples/imaging_demo.py [hunk 2]
```diff
@@ -62,57 +56,51 @@
...
+idx = np.argwhere(np.isin(cal_vis.meta["isc"], isc_10_7)).ravel()
 
 ###############################################################################
-# Create an ``xrayvsion`` Visibilities object
+# Slice the visibilities to detectors 10 - 7
 
-stix_vis = Visibilities(
-    cal_vis.obsvis[idx], cal_vis.u[idx], cal_vis.v[idx], amplitude_uncertainty=cal_vis.amplitude_error[idx]
-)
+vis10_7 = cal_vis[idx]
```
Comment (DanRyanIrish) [inline examples/imaging_demo.py:103]: This should use `Visibilities.index_by_label`

Comment (samaloney) [reply examples/imaging_demo.py:103]: So the label are 10a, 10b, etc so was easier to directly slice here. Also doesn't seem to like using a list `vis.index_by_label(['10a', '10b'])` doesn't work but `vis.index_by_label('10a', '10b')` does I know why but is this the API we want?

File: examples/imaging_demo.py (hunk 1 above)
Comment (samaloney) [reply examples/imaging_demo.py:234]: Currently `vis_clean` makes a map assuming helioprojective which in this case is not correct.

File: examples/imaging_demo.py (hunk 2 above)
Comment (samaloney) [reply examples/imaging_demo.py:103]: I've opened TCDSolar/xrayvision/issues/71 to track this.

File: examples/imaging_demo.py (hunk 1 above)
Comment (samaloney) [reply examples/imaging_demo.py:234]: Essentially once TCDSolar/xrayvision/issues/56 is done issue is resolved this won't be necessary

**Pull request merged**